src/model/storage.py

```diff
@@ -72,7 +74,9 @@ def _load_model(model_id):
     response = requests.get(f"{app.config['MODEL_WAREHOUSE_API']}/models/{model_id}")
 
     if response.status_code == 404:
-        raise KeyError(f"No model with id {model_id}")
+        raise ModelNotFound(f"No model with id {model_id}")
```

Comment:
See above. Should the message for 401 not be distinct from 403?

Reply:
Well... yes, I took a bit of a shortcut here. 401 should only occur for invalid/expired JWTs, but you're right, the difference shouldn't be hidden, so I'll make an update to distinguish them.